src/schnetpack/data/datamodule.py

```diff
@@ -268,21 +268,60 @@ def _load_partitions(self):
                 self.train_idx = S["train_idx"].tolist()
                 self.val_idx = S["val_idx"].tolist()
                 self.test_idx = S["test_idx"].tolist()
-                if self.num_train and self.num_train != len(self.train_idx):
-                    logging.warning(
-                        f"Split file was given, but `num_train ({self.num_train})"
-                        + f" != len(train_idx)` ({len(self.train_idx)})!"
-                    )
-                if self.num_val and self.num_val != len(self.val_idx):
-                    logging.warning(
-                        f"Split file was given, but `num_val ({self.num_val})"
-                        + f" != len(val_idx)` ({len(self.val_idx)})!"
-                    )
-                if self.num_test and self.num_test != len(self.test_idx):
-                    logging.warning(
-                        f"Split file was given, but `num_test ({self.num_test})"
-                        + f" != len(test_idx)` ({len(self.test_idx)})!"
-                    )
+
+                total_size = len(self.dataset)  # Get dataset size
+
+                # If num_train/num_val/num_test are fractions, convert them to absolute values
+                if isinstance(self.num_train, float) and self.num_train <= 1:
+                    expected_train_size = int(self.num_train * total_size)
+                else:
+                    expected_train_size = self.num_train
+
+                if isinstance(self.num_val, float) and self.num_val <= 1:
+                    expected_val_size = int(self.num_val * total_size)
+                else:
+                    expected_val_size = self.num_val
+
+                if isinstance(self.num_test, float) and self.num_test <= 1:
+                    expected_test_size = int(self.num_test * total_size)
+                else:
+                    expected_test_size = self.num_test
```

Comment:
remove this part and replace with updating `self.num_xxx` with absolute values. (See my comment in line 334)